tests/main/snap-quota-cpu/task.yaml

```diff
@@ -0,0 +1,129 @@
+summary: Functional test for cpu quota-related snap commands.
+
+details: |
+  Functional test for snap cpu quota group commands ensuring that they are 
+  effective in practice.
+
+# these systems do not support quota groups due to their old systemd versions,
+# we do have a check to do for these systems, but that is just to ensure that
+# we don't allow using quota groups, and that check is done in the snap-quota
+# spread instead
+systems:
+  - -ubuntu-14.04-*
+  - -centos-7-*
+  - -amazon-linux-2-*
+  - -ubuntu-16.04-*
+  - -ubuntu-core-16-*
+
+prepare: |
+  snap install test-snapd-stressd --edge --devmode
+  snap set system experimental.quota-groups=true
+  tests.cleanup defer snap unset system experimental.quota-groups
+
+execute: |
+  echo "Starting service and verifying cpu usage"
+  snap start test-snapd-stressd.stress-sc
+
+  # helper function that has retry logic to allow for cpu usage to stabilize
+  function ensure_cpu_usage_higher_than {
+    n=0
```

Comment:
(nitpick) one nice patter for shell functions is to do something like:
```
local name="$1"
local percent="$2"
...
```
to make it more obvious what is expected to get passed and also to avoid confusion with things like the `awk '{print $9*100}'` that is not coming from the commandline but on a quick naive look made me wonder why we pass $9 into the function (probably not a great example as only silly people will read it like this but hopefully it illustrate the point that $[0-9] is used all over the place in shell and it might be cofusing to see what is passed in or not).

Reply:
Really good point, makes it a lot clearer in regards to input! Thank you for pointing this out!